### PR TITLE
Optimisation, reduce number of backordered items iterated 

### PR DESCRIPTION
**Pull request by alex-handley**
Background: When there is a large number of backordered items we are seeing bottleneck/memory issue on this method because of the number of objects created.

To improve this I have changed the code so that it queries stock location in AR.
This was changed to work in this way back in 2013 - https://github.com/spree/spree/issues/3066

Does anyone have any feedback or ideas on negative affects this could cause?

### CODE REVIEW

File: core/app/models/spree/inventory_unit.rb [hunk 1]
```diff
@@ -62,13 +62,9 @@ class InventoryUnit < Spree::Base
...
     def self.backordered_for_stock_item(stock_item)
-      # This was refactored from a simpler query because the previous
-      # implementation led to issues once users tried to modify the objects
-      # returned. That's due to ActiveRecord `joins(shipment: :stock_location)`
-      # only returning readonly objects
-      backordered_per_variant(stock_item).select do |unit|
-        unit.shipment.stock_location == stock_item.stock_location
-      end
+      backordered_per_variant(stock_item)
+        .where("spree_shipments.stock_location_id = ?",
```
Comment (Senjai) [inline core/app/models/spree/inventory_unit.rb]: :+1: Now that this doesnt return an array, but a relation, we could probably move it into a scope.

Might be cleaned up by `where(shipment: { stock_location_id: stock_item.stock_location_id })`

**Comment by alex-handley** [comment]
@Senjai Good idea, I always forget about the nested where syntax!

I have updated the PR.

**Comment by cbrunsdon** [comment]
:+1: looks good to me, thanks!

**Comment by athal7** [comment]
:+1:

**Pull request merged**